src/main/java/com/guciowons/shoppingbasket/Basket/BasketService.java

```diff
@@ -1,18 +1,18 @@
 package com.guciowons.shoppingbasket.Basket;
 
-import com.guciowons.shoppingbasket.Product.ProductDao;
+import com.guciowons.shoppingbasket.Product.ProductClient;
 import org.springframework.stereotype.Service;
 
 @Service
 public class BasketService {
-    private final ProductDao productDao;
     private final BasketSummarizer basketSummarizer;
     private final BasketDao basketDao;
+    private final ProductClient productClient;
```

Comment:
Tutaj raczej powinnieneś wrzuciś `ProductService` -- serwisy powinny komunikować się między sobą i nie udostępniać na zewnątrz swoich zależności, to tworzy tak zwanny spaghetti code gdzie wszystko jest połączone ze wszystkim i potem ciężko to odplątać.

Jeśli masz dwa typy serwisów obsługujących różne ficzery to one powinny się komunikować ze sobą tylko na poziomie serwisów -- Repository, Providery itd itp powinny być za nimi schowane.